packages/core/src/repository/repository.adapter.ts

```diff
@@ -0,0 +1,20 @@
+import { Repository } from './repository.service.js';
+
+/**
+ * Interface for repository platform adapters
+ */
+export interface RepositoryAdapter {
```

Comment:
Find a better name, it is not a repository adapter but an adapter handle repository service implementation